src/Orchard.Environment.Extensions/Folders/ManifestParsers/JsonManifestParser.cs

```diff
@@ -0,0 +1,204 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Newtonsoft.Json.Linq;
+using Orchard.Environment.Extensions.Models;
+using Orchard.FileSystem;
+using Orchard.Localization;
+
+namespace Orchard.Environment.Extensions.Folders.ManifestParsers
+{
+    public class JsonManifestParser : IManifestParser
+    {
+        private readonly IClientFolder _clientFolder;
+
+        public JsonManifestParser(IClientFolder clientFolder)
+        {
+            _clientFolder = clientFolder;
+            T = NullLocalizer.Instance;
+        }
+
+        public Localizer T { get; set; }
+
+        public string Extension => ".json";
+
+        public ExtensionDescriptor GetDescriptorForExtension(string path, string extensionId, string extensionType, string manifestText)
+        {
+            dynamic project = ReadJsonFile(path, extensionId, "project.json");
```

Comment:
Don't read the project.json file like this... use the Project.Load or something like that object.